pkg/verification/service/verification_service_test.go

```diff
@@ -896,41 +896,122 @@ func (s *TestVerificationServiceSuite) TestVerifyPhoneCode() {
-					toolchainv1alpha1.UserVerificationExpiryAnnotationKey:     now.Add(10 * time.Second).Format(verificationservice.TimestampLayout),
-				},
-				Labels: map[string]string{
-					toolchainv1alpha1.UserSignupUserPhoneHashLabelKey: "+1NUMBER",
-				},
+	s.T().Run("captcha configuration ", func(t *testing.T) {
+		tests := map[string]struct {
+			activationCounterAnnotationValue string
+			captchaScoreAnnotationValue      string
+			allowLowScoreConfiguration       bool
+			expectedErr                      string
+		}{
+			"captcha score below required score but it's a reactivation": {
+				activationCounterAnnotationValue: "2",   // user is reactivating
+				captchaScoreAnnotationValue:      "0.5", // and captcha score is low
+				allowLowScoreConfiguration:       true,
 			},
-			Spec: toolchainv1alpha1.UserSignupSpec{
-				Username: "sbryzak@redhat.com",
+			"captcha score below required score but it's not a reactivation": {
+				activationCounterAnnotationValue: "1",   // first time user
+				captchaScoreAnnotationValue:      "0.5", // and captcha score is low
+				allowLowScoreConfiguration:       true,
+				expectedErr:                      "verification failed: verification is not available at this time",
+			},
+			"activation counter is invalid and captcha score is low": {
+				activationCounterAnnotationValue: "x",   // something wrong happened
+				captchaScoreAnnotationValue:      "0.5", // and captcha score is low
+				allowLowScoreConfiguration:       true,
+				expectedErr:                      "verification failed: verification is not available at this time",
+			},
+			"activation counter is invalid and captcha score is ok": {
+				activationCounterAnnotationValue: "x",   // something wrong happened
+				captchaScoreAnnotationValue:      "0.6", // but captcha score is ok
+				allowLowScoreConfiguration:       true,
+			},
+			"allow low score disabled - captcha score below required score and it's a reactivation": {
+				activationCounterAnnotationValue: "2",   // user is reactivating
+				captchaScoreAnnotationValue:      "0.5", //  captcha score is low
+				allowLowScoreConfiguration:       false,
+				expectedErr:                      "verification failed: verification is not available at this time",
+			},
+			"allow low score disabled - captcha score below required score and it's not a reactivation": {
+				activationCounterAnnotationValue: "1",   // first time user
+				captchaScoreAnnotationValue:      "0.5", //  captcha score is low
+				allowLowScoreConfiguration:       false,
+				expectedErr:                      "verification failed: verification is not available at this time",
+			},
+			"allow low score disabled - captcha score ok": {
+				activationCounterAnnotationValue: "1",   // first time user
+				captchaScoreAnnotationValue:      "0.6", //  captcha score is ok
+				allowLowScoreConfiguration:       false,
+			},
+			"no score annotation": {
```

Comment:
Let's also add a test for the case with no activation counter annotation set at all.

Reply:
good point! done , please see https://github.com/codeready-toolchain/registration-service/pull/379/commits/c09beb3d59e404b08a99285b2d7e6437a5b4b3df